File: src/main/java/com/company/erp/features/chats/dto/StreamTokenDto.java
package com.company.erp.features.chats.dto;

import java.time.Instant;

/**
 * Short-lived Stream Video user token. Mobile clients exchange this for an
 * authenticated session against the Stream SDK, which carries the actual
 * audio/video traffic. The signalling ceremony (RINGING / ANSWERED / ENDED)
 * stays on our side over STOMP.
 */
public record StreamTokenDto(
        String token,
        String apiKey,
        String userId,
        Instant expiresAt
) {
}

File: app/Features/Chat/Dto/StreamTokenDto.php
<?php

namespace App\Features\Chat\Dto;

use DateTimeInterface;

/**
 * Port of the Spring StreamTokenDto record — a short-lived Stream Video user
 * token the mobile SDK exchanges for an authenticated media session.
 */
final class StreamTokenDto
{
    /**
     * @return array<string, mixed>
     */
    public static function from(string $token, string $apiKey, string $userId, DateTimeInterface $expiresAt): array
    {
        return [
            'token' => $token,
            'apiKey' => $apiKey,
            'userId' => $userId,
            'expiresAt' => $expiresAt->format('Y-m-d\TH:i:s\Z'),
        ];
    }
}
